src/Command/MarkMigrated.php

```diff
@@ -150,4 +159,95 @@ protected function markAllMigrated($path)
         }
         $adapter->commitTransaction();
     }
+
+    /**
+     * Decides based on $input if it will try to mark more than one migration
+     *
+     * @param InputInterface $input
+     * @return boolean
+     */
+    protected function isBatchMarking(InputInterface $input)
+    {
+        $version = $input->getArgument('version');
+
+        return $this->isAllVersion($version) ||
+            $this->isSkipTo($input) ||
+            $this->isUpTo($input);
+    }
+
+    /**
+     * Decides based on $input which versions it should mark as migrated
+     *
+     * @param InputInterface $input
+     * @return array of versions that should be marked as migrated
+     */
+    protected function versionsToMark(InputInterface $input)
+    {
+        $version = $input->getArgument('version');
+        $migrations = $this->getManager()->getMigrations();
+        $versions = array_keys($migrations);
+
+        if ($this->isAllVersion($version)) {
+            return $versions;
+        }
+
+        $lengthIncrease = $this->isSkipTo($input) ? 0 : 1;
+        $index = array_search($version, $versions);
+
+        if (FALSE === $index) {
```

Comment:
I have the same feeling than you.
I find it easier to read. Plus, with a strict equality check, it's hard to do it by mistake.

But I know there are supporter for the ``value == $var`` type of writing.